tests/backends/database/test_es.py

```diff
@@ -0,0 +1,128 @@
+"""Tests for Ralph es database backend"""
+
+import json
+import sys
+from collections.abc import Iterable
+from io import BytesIO, StringIO
+from pathlib import Path
+
+from elasticsearch import Elasticsearch
+from elasticsearch.helpers import bulk
+
+from ralph.backends.database.es import ESDatabase
+from ralph.defaults import APP_DIR, HISTORY_FILE
+
+from tests.fixtures.backends import ES_TEST_HOSTS, ES_TEST_INDEX
+
+
+def test_es_database_instanciation(es):
+    """Test the ES backend instanciation"""
+    # pylint: disable=invalid-name,unused-argument,protected-access
+
+    assert ESDatabase.name == "es"
+
+    database = ESDatabase(
+        hosts=ES_TEST_HOSTS,
+        index=ES_TEST_INDEX,
+    )
+
+    # When running locally host is 'elasticsearch', while it's localhost when
+    # running from the CI
+    assert any(
+        (
+            "http://elasticsearch:9200" in database._hosts,
+            "http://localhost:9200" in database._hosts,
+        )
+    )
+    assert database.index == ES_TEST_INDEX
+    assert isinstance(database.client, Elasticsearch)
+
+
+def test_to_documents_method(es):
```

Comment:
IMO a private method should not be test. You will test it through the public one using it.

Reply:
Isolated a piece of code in a private method is to make it easily reusable AND testable. I'm not sure it shouldn't be tested.

Reply:
> solated a piece of code in a private method is to make it easily reusable and testable
I disagree, why not implementing a public method in that case ? 

You are testing twice.

Reply:
A unit test is a client of the object you are testing. Never a client will access your private method, if python implement real private method you will see how difficult it is to test a private method.

Reply:
Ok, I've made this method public instead.